Moved 8m surefire config for page cache to root pom.xml

tinwelint opened the pull request:
so that it will apply to all testing. Tests that want bigger page cache
will have to explicitly specify that.

tinwelint committed:
Moved 8m surefire config for page cache to root pom.xml

so that it will apply to all testing. Tests that want bigger page cache
will have to explicitly specify that.

chrisvest commented:
@tinwelint I think we can safely lower the lower-bound on the `GraphDatabaseSettingsTest.mustHaveReasonableDefaultPageCacheMemorySizeInBytes` to make that one pass.

I don't know what's wrong with the other test. I think it should be investigated.

tinwelint committed:
Fixes issues defragmenting rel groups for double-unit records

where some batches might have one or more trailing relationship group records
not prepared correctly which would leave groups which in reality required a
secondary record unit without one.

Now the reader will not break apart groups from any given node, but always
keep all groups for any given node in one batch (potentially resulting in
slightly bigger batches, which is perfectly fine).

tinwelint committed:
Reduced expected amount of lowest default memory to have configured in test

tinwelint commented:
@chrisvest made that change. The other test was actually a real issue, which is now fixed as part of this PR, see b6d92e25c30e32aff92c0521b2c4da52f109a635

chrisvest commented on community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/staging/ProducerStep.java:
```diff
@@ -56,6 +56,7 @@ public void run()
                 }
                 catch ( Throwable e )
                 {
+                    e.printStackTrace();
```
Are you sure?

chrisvest commented on community/kernel/src/test/java/org/neo4j/unsafe/impl/batchimport/RelationshipGroupDefragmenterTest.java:
```diff
@@ -213,6 +213,8 @@ private void verifyGroupsAreSequentiallyOrderedByNode()
             }
 
+//            System.out.println( "read:" + groupRecord );
+
```
This can go away

tinwelint commented:
@chrisvest a bit sloppy, removed those two lines
